hack/make-rules/helpers/go2make/go2make.go

```diff
@@ -0,0 +1,164 @@
+)
+
+var flImport = pflag.StringSlice("import", nil, "packages to process (may be specified multiple times)")
+var flPrune = pflag.StringSlice("prune", nil, "sub-packages to prune (recursive) (may be specified multiple times)")
+var flDebug = pflag.BoolP("debug", "d", false, "enable debugging output")
+
+func main() {
+	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
+	pflag.Parse()
+
+	debug("PWD", getwd())
+
+	build.Default.BuildTags = []string{"ignore_autogenerated"}
+	build.Default.UseAllFiles = false
+	for _, in := range *flImport {
+		if strings.HasSuffix(in, "/...") {
+			// Recurse.
+			debug("starting", in)
+			pkgName := strings.TrimSuffix(in, "/...")
+			if err := WalkPkg(pkgName, visitPkg); err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(1)
+			}
+		} else {
+			// Import one package.
+			if err := saveImport(in); err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(2)
+			}
+		}
+	}
+}
+
+func debug(items ...interface{}) {
+	if *flDebug {
+		x := []interface{}{"DBG:"}
+		x = append(x, items...)
+		fmt.Println(x...)
+	}
+}
+
+func visitPkg(importPath, absPath string) error {
+	debug("visit", importPath)
+	if prune(importPath) {
+		return ErrSkipPkg
+	}
+	return saveImport(importPath)
+}
+
+func prune(pkgName string) bool {
+	for _, pr := range *flPrune {
+		if pr == pkgName {
+			return true
+		}
+	}
+	return false
+}
+
+// cache keeps track of which packages we have already loaded.
+var cache = map[string]*build.Package{}
+
+func saveImport(pkgName string) error {
+	if cache[pkgName] != nil {
+		return nil
+	}
+	pkg, err := loadPackage(pkgName)
+	if err != nil {
+		return err
+	}
+	cache[pkgName] = pkg
+
+	debug("recurse", pkgName)
+	defer func() { debug("done   ", pkgName) }()
+	if !pkg.Goroot && (len(pkg.GoFiles)+len(pkg.Imports) > 0) {
+		// Process deps of this package before the package itself.
+		for _, impName := range pkg.Imports {
+			if impName == "C" {
+				continue
+			}
+			debug("depends on", impName)
+			saveImport(impName)
+		}
+
+		// Emit a variable for each package.
+		var buf bytes.Buffer
+		buf.WriteString(pkgName)
+		buf.WriteString(" := ")
+
+		// Packages depend on their own directories.
+		buf.WriteString(pkg.Dir)
+
+		// Sort and de-dup all inputs.
+		buf.WriteString(" $(sort")
+
+		// Packages depend on their own files.
+		for _, file := range pkg.GoFiles {
+			buf.WriteString(" ")
+			if pkg.Dir != "." {
+				buf.WriteString(pkg.Dir)
+				buf.WriteString("/")
+			}
+			buf.WriteString(file)
+		}
+		for _, impName := range pkg.Imports {
+			pkg := cache[impName]
+			if pkg == nil || pkg.Goroot {
+				continue
+			}
+			buf.WriteString(" $(")
+			buf.WriteString(impName)
+			buf.WriteString(")")
+		}
+		buf.WriteString(")")
+		fmt.Println(buf.String())
+	}
+	return nil
+}
+
+func loadPackage(pkgName string) (*build.Package, error) {
+	debug("load", pkgName)
+	pkg, err := build.Import(pkgName, getwd(), 0)
+	if err != nil {
+		// We can ignore NoGoError.  Anything else is real.
```

Comment:
Is `pkg` nil on that error?

Reply:
No. NoGoError returns a valid value *and* an error.